### ISSUE TITLE
Twitter-Like-System-PHP– Cross-Site Scripting (XSS)

### ISSUE THREAD

**Issue by bestshow** [iSimar/Twitter-Like-System-PHP#4]
Product: Twitter-Like-System-PHP
Download: https://github.com/iSimar/Twitter-Like-System-PHP
Vunlerable Version: latest version
Tested Version: latest version
Author: ADLab of Venustech

Advisory Details:
A Cross-Site Scripting (XSS) was discovered in “Twitter-Like-System-PHP latest version”, which can be exploited to execute arbitrary code.
The vulnerability exists due to insufficient filtration of user-supplied data in the “hashtag” HTTP GET parameter passed to the “Twitter-Like-System-PHP-master/hashtag.php” URL. An attacker could execute arbitrary HTML and script code in a browser in the context of the vulnerable website.
The exploitation example below uses the "alert()" JavaScript function to see a  pop-up messagebox:
Poc:
http://localhost/.../Twitter-Like-System-PHP-master/hashtag.php?hashtag=%22%3E%3C/script%3E%3Cscript%3Ealert(1);%3C/script%3E

**Comment by bestshow** [comment]
Excuse me, is there anyone dealing with this issue?

**Comment by Phumix** [comment]
I never experienced this when I used it. I found a link with some good tips on how to prevent it, have a look:
http://stackoverflow.com/questions/1996122/how-to-prevent-xss-with-html-php

Also, it's not safe to use this repo anymore because it's using MySQL instead of MySQLi or PDO. 
The mysql_* functions has been deprecated as of 5.5.0 and it's highly recommended to use something else. Last time I used this repo, I re-wrote most of the code and changed it all to MySQLi functions.
